src/gui/Refine2DResultsPanel.cpp

```diff
@@ -60,41 +60,45 @@ void Refine2DResultsPanel::OnClassumLeftClick(wxMouseEvent& event) {
 
         ClassificationSelection* current_selection = &refinement_package_asset_panel->all_classification_selections.Item(SelectionManagerListCtrl->ReturnCurrentSelectionOriginalArrayPosition( ));
 
-        // ok toggle the effective selection..
+        // Need to make sure the event's ID is within the bounds of the images, i.e. the classes in this MRC
+        // Failure to do this results in crash when non-existent index access is attempted.
+        if ( event.GetId( ) <= current_selection->number_of_classes ) {
```

Comment:
So this only needed to be bounded from above?

Is wx indexing from 1 (why <= rather than <)

Reply:
> Is wx indexing from 1 (why <= rather than <)

wxWidgets indexing is still from 0, but the image_is_selected bool array technically uses 1 indexing as its size is set to be `ReturnNumberOfSlices() + 1`. I think this was done for keeping consistent with the 1-indexing of the MRC type.

> So this only needed to be bounded from above?

The DisplayNotebookPanel for this DisplayPanel allocates enough space across all rows/columns for images of the same size (for me this was 3 rows, 4 columns by default). Clicks with wx then use these as a separate event ID, and when `ToggleImageSelected` gets called, it fails an assert at line 1809 of DisplayPanel.cpp:
```
MyDebugAssertTrue(wanted_image > 0 && wanted_image <= ReturnNumberofImages( ), "Trying to select an image that doesn't exist (%li)", wanted_image);
```
 and crashes because the selected ID can be greater than the actual number of images (for example, having 10 classes instead of 12 leaves 2 spaces that can be clicked and cause a crash).

And from my experimentation and reading, it seemed the event ID would always be greater than 0, so it made sense to only bind from above -- though maybe binding both directions is technically safer.